### PR TITLE
[ticket/14919] Do not directly use globals in acp_extensions

### PR DESCRIPTION
**Pull request by javiexin**
PHPBB3-14919

Checklist:

- [ ] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [ ] Tests pass
- [ ] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [ ] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket:

https://tracker.phpbb.com/browse/PHPBB3-14919

### CODE REVIEW

File: phpBB/includes/acp/acp_extensions.php [hunk 1]
```diff
@@ -22,35 +22,37 @@
 class acp_extensions
 {
 	var $u_action;
+	var $tpl_name;
```
Comment (javiexin) [inline phpBB/includes/acp/acp_extensions.php]: Consistency: in other acp modules, the situation is:
- $u_action is mostly declared as var, with some instances as public (very few)
- $tpl_name is used but never declared
- $page_title is used but never declared
What should we do?  Keep consistency (do as above), or do it "right" (declare all three as public)?  Or any other alternative?

File: phpBB/includes/acp/acp_extensions.php [hunk 2]
```diff
@@ -22,35 +22,37 @@
...
 
-	private $db;
```
Comment (javiexin) [inline phpBB/includes/acp/acp_extensions.php]: Declared but never used

File: phpBB/includes/acp/acp_extensions.php [hunk 3]
```diff
@@ -22,35 +22,37 @@
...
 	private $config;
 	private $template;
 	private $user;
 	private $cache;
 	private $log;
 	private $request;
+	private $php_dispatcher;
```
Comment (javiexin) [inline phpBB/includes/acp/acp_extensions.php]: Used without declaration

File: phpBB/includes/acp/acp_extensions.php [hunk 4]
```diff
@@ -22,35 +22,37 @@
...
+	private $ext_manager;
```
Comment (javiexin) [inline phpBB/includes/acp/acp_extensions.php]: Used as global, not declared or assigned

File: phpBB/includes/acp/acp_extensions.php (hunk 1 above)
Comment (Nicofuma) [inline phpBB/includes/acp/acp_extensions.php]: declare all 3

Comment (javiexin) [inline phpBB/includes/acp/acp_extensions.php]: As var, or public?

Comment (Nicofuma) [inline phpBB/includes/acp/acp_extensions.php]: var to be consistent I guess...

Comment (javiexin) [inline phpBB/includes/acp/acp_extensions.php]: To be really consistent we should only declare u_action as var :)
I will make another commit with this change.

Comment (javiexin) [inline phpBB/includes/acp/acp_extensions.php]: To be really consistent we should only declare u_action as var :)
I will make another commit with this change.

**Comment by Nicofuma** [comment]
This is gonna need a separate PR for 3.2 with the merge conflict fix (see the first 4 points here: https://area51.phpbb.com/docs/dev/31x/development/git.html#merging-to-3-1-x-and-master-with-different-patches)

**Comment by javiexin** [comment]
Sorry, but I am not able to see the merge conflict...  
What I have seen is that the call to create metadata objects does not carry a $template param any longer, but the call to output template vars does.  These changes are easy to do, and in fact, I have them ready.
But how do I do this from Git perspective?  New local branch off of 3.2, and then push that and new PR that?  Given that the code is different, this looks like the most appropriate way, but what should be the branch name so that the merge goes well?

**Comment by javiexin** [comment]
Ok, I have tried, but I am not managing to do it properly.
I have created a gist with the 3.2 version of this file.
https://gist.github.com/javiexin/022697e84a613665d334f1938d3acc88
Hope this is ok, but I don't know how to properly do the second PR without the conflicts...

**Comment by javiexin** [comment]
The PR for ticket/15087 includes the 3.2 version of this PR. I have removed the above gist, now obsolete.

**Pull request merged**